### PR TITLE
Added larch normalization scheme to postprocessing

### PR DESCRIPTION
**Pull request by CharlesC30**
Created new class for larch normalization scheme in operations module. Also added testing for new operator in tutorial notebook.

### CODE REVIEW

**Comment by CharlesC30** [comment]
@x94carbone

File: aimmdb/postprocessing/operations.py [hunk 1]
```diff
@@ -350,6 +352,47 @@ def _process_data(self, df):
...
+        References a list of columns in the DataFrameClient (these are the
+        "y-axes"). Default is ["mu"].
+    """
+
+    def __init__(self, *, x_column="energy", y_columns=["mu"]):
+        self.x_column = x_column
+        self.y_columns = y_columns
+
+    @staticmethod
+    def flatten(group: xafsgroup):
```
Comment (x94carbone) [inline aimmdb/postprocessing/operations.py:374]: Should be a "private" method, i.e. `_flatten` not `flatten`.

**Review by x94carbone — CHANGES_REQUESTED** [review]
Looks really solid, thank you. Few changes are requested!

**Review by x94carbone — CHANGES_REQUESTED** [review]
@CharlesC30 looks good, but I still think that `_process_data` could use some documentation. It would just be nice to know what Larch is doing to some degree. Actually, it might be best to put this in the doctoring for `NormalizeLarch`. Don't waste more than a day trying to figure out exactly what it's doing, worst case we just use it as is and don't worry about it 😁

**Comment by CharlesC30** [comment]
@x94carbone I added an explanation of the normalization process in larch to the `NormalizeLarch` docstring. Some of it is copied from the larch source code then modified slightly for how we are using it here.

**Comment by CharlesC30** [comment]
Also if we wanted to we could add more parameters to this class to interface with larch more. 
For example there could be a keyword argument `e0` (default equal to `None`), that is also given to the larch `pre_edge` function during normalization. 
This could also be done for other keyword arguments in the `pre_edge` function. Though I am not sure if this is how we want these operators to work.

**Comment by x94carbone** [comment]
@CharlesC30 perhaps just add a ``**kwargs`` arg to the init or something like that?

**Comment by CharlesC30** [comment]
> @CharlesC30 perhaps just add a `**kwargs` arg to the init or something like that?

good idea! added

**Review by x94carbone — APPROVED** [review]
Looks good! @CharlesC30 is this still a work in progress or are you good to merge?

**Comment by CharlesC30** [comment]
> I like this. That would be part of forcing the Larch normalization parameters, right?

@x94carbone yeah exactly! By default larch calculates most parameters itself but we could explicitly pass things that are already known.

> Give it some thought, let me know what you think! The general idea is that the operator would take these types of parameters during instantiation. Then the `__call__` method just takes what's being operated on.

Okay I think it's not a bad idea then. I was looking at the other operators though and think maybe we should use the same approach as the `StandardizeGrid` operator. Basically provide a dict keyword argument to the `__init__` that contains all the larch `pre_edge` kwargs, then link to the `pre_edge` documentation in our docstring.

**Comment by x94carbone** [comment]
@CharlesC30 Yep this is really good, great work, thank you! Basically exactly what I was looking for. I think this PR is good to go unless you have any other suggestions.

**Comment by CharlesC30** [comment]
@x94carbone All good with me! I think it's okay to merge

**Pull request merged**